src/components/App.tsx

```diff
@@ -408,6 +415,64 @@ class App extends React.Component<AppProps, AppState> {
     this.actionManager.registerAction(createRedoAction(this.history));
   }
 
+  setCustomType = (customType: string) => {
+    this.setState({
+      activeTool: { ...this.state.activeTool, type: "custom", customType },
+    });
+  };
+
+  renderCustomElement = (coords: { x: number; y: number }) => {
+    if (this.state.activeTool.type !== "custom") {
+      return;
+    }
+    const config = getCustomElementConfig(
+      this.props.customElementsConfig,
+      this.state.activeTool.customType,
+    );
+    if (!config) {
+      return;
+    }
+    const [gridX, gridY] = getGridPoint(
+      coords.x,
+      coords.y,
+      this.state.gridSize,
+    );
+
+    const width = config.width || 40;
+    const height = config.height || 40;
+    const customElement = newCustomElement(this.state.activeTool.customType, {
+      type: "custom",
+      x: gridX - width / 2,
+      y: gridY - height / 2,
+      strokeColor: this.state.currentItemStrokeColor,
+      backgroundColor: this.state.currentItemBackgroundColor,
+      fillStyle: this.state.currentItemFillStyle,
+      strokeWidth: this.state.currentItemStrokeWidth,
+      strokeStyle: this.state.currentItemStrokeStyle,
+      roughness: this.state.currentItemRoughness,
+      opacity: this.state.currentItemOpacity,
+      strokeSharpness: this.state.currentItemLinearStrokeSharpness,
+      width,
+      height,
+      locked: false,
+    });
+
+    const unbind = this.scene.addCallback(() => {
+      const customElementConfig = getCustomElementConfig(
+        this.props.customElementsConfig,
+        customElement.customType,
+      );
+      if (customElementConfig && customElementConfig.onCreate) {
+        customElementConfig.onCreate(customElement);
+      }
+    });
+    this.scene.replaceAllElements([
+      ...this.scene.getElementsIncludingDeleted(),
+      customElement,
+    ]);
+    unbind();
```

Comment:
doing this bind/unbind thing is either 1) very unnecessary (current case), you can just call the `onCreate()` call on line 473 directly, or 2) very unsafe (in the future if the `replaceAllElements` becomes async)